### PR TITLE
feat: replace webpack with vite

### PR DESCRIPTION
**Pull request by sehilyi**
This replaces the webpack with vite.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2,21 +2,19 @@
 <html lang="en">
     <head>
         <meta charset="utf-8" />
-        <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+        <link rel="icon" href="/src/favicon.ico" />
         <meta name="viewport" content="width=device-width, initial-scale=1" />
-        <meta name="theme-color" content="lightgray" />
-        <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
-        <link rel="stylesheet" href="https://unpkg.com/higlass@1.11.3/dist/hglib.css">
+        <link rel="stylesheet" href="https://unpkg.com/higlass@1.12/dist/hglib.css">
```
Comment (manzt) [inline index.html]: We should have an import for`hglib.css` in `src/index.jsx` rather than relying on a CDN. This can easily go out of date, leading to conflicting styles, when the version in the package.json is bumped.

Should be something like 

```javascript
import "higlass/dist/hglib.css";
```

File: index.html [hunk 2]
```diff
@@ -2,21 +2,19 @@
...
         <script src="https://cdn.tailwindcss.com"></script>
         <style>
             /* To resolve style conflicts with Tailwind */
             .gosling-component { height: auto !important; }
         </style>
-        <title>Gosling.js React Example</title>
+        <title>Gosling React Vite Example</title>
```
Comment (manzt) [inline index.html]: I think it's fine to leave as "Gosling React Example". Vite is just a build tool.

File: package.json [hunk 3]
```diff
@@ -1,51 +1,43 @@
 {
-  "name": "gosling-react-template",
+  "name": "gosling-react-vite-example",
```
Comment (manzt) [inline package.json]: Again, I think that it's we  keep  this as as `gosling-react-template`. We are just migrating our recommended build tool (i.e., we didn't call it gosling-react-webpack-example). The only thing that changed was the packaging, not any of the source code (it's still React).

File: vite.config.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { defineConfig } from 'vite';
+import reactRefresh from '@vitejs/plugin-react-refresh';
+
+// https://vitejs.dev/config/
+export default defineConfig({
+    build: { target: 'esnext' },
+    optimizeDeps: {
+        include: ['gosling.js']
```
Comment (manzt) [inline vite.config.js]: why is this necessary?

File: package.json (hunk 3 above)
Comment (sehilyi) [reply package.json]: I wanted to make it clearer, but I agree. Will remove `vite-`

File: index.html (hunk 1 above)
Comment (sehilyi) [reply index.html]: Ah, you are right. I had the issue with the outdated CSS which require quite time to figure out what was wrong..

File: vite.config.js (hunk 4 above)
Comment (sehilyi) [reply vite.config.js]: It is not necessary. It was left over after testing something.

**Comment by sehilyi** [comment]
@ThHarbig Just a heads-up that you will need to use `yarn` (https://yarnpkg.com/getting-started/install) (instead of npm) to install all dependencies and `yarn start` to open the demo running.

https://github.com/gosling-lang/gosling-react-example#using-this-repository
